### PR TITLE
Search for approved invesetment requests and update state

### PR DESCRIPTION
**Pull request by DavidGershony**

### CODE REVIEW

File: src/Angor/Client/Pages/Signatures.razor [hunk 1]
```diff
@@ -172,43 +157,99 @@
...
                     pendingSignature.AmountToInvest = investorTrx.Outputs.AsIndexedOutputs().Skip(2).Take(investorTrx.Outputs.Count - 3) //Todo get the actual outputs with taproot type
                         .Sum(_ => _.TxOut.Value);
-                    
-                    StateHasChanged();
                 }
                 catch (Exception e)
                 {
                     Console.WriteLine(e);
-                    Console.WriteLine(pendingSignature.TransactionHex);
+                    _Logger.LogDebug(pendingSignature.TransactionHex);
```
Comment (dangershony) [inline src/Angor/Client/Pages/Signatures.razor:164]: There should be more text around logs, later we wont be able to know where this came from.

File: src/Angor/Shared/DerivationOperations.cs [hunk 2]
```diff
@@ -216,6 +216,22 @@ public Key DeriveProjectNostrPrivateKey(WalletWords walletWords, int index)
...
     }
+    
+    public async Task<Key> DeriveProjectNostrPrivateKeyAsync(WalletWords walletWords, int index)
+    {
+        // founder key is derived from the path m/5'
+        ExtKey extendedKey = GetExtendedKey(walletWords);
+
+        var path = $"m/44'/1237'/{index}/0/0";
+
+        var task = Task.Run(() => extendedKey.Derive(new KeyPath(path)));
```
Comment (dangershony) [inline src/Angor/Shared/DerivationOperations.cs:227]: this doesn't help in anyway to make it faster, drop the Task here if you ask me it is just confusing.

**Review by dangershony — APPROVED** [review]

File: src/Angor/Shared/DerivationOperations.cs (hunk 2 above)
Comment (DavidGershony) [reply src/Angor/Shared/DerivationOperations.cs:227]: Not faster but the UI doesn't freeze when it runs in a task

Comment (dangershony) [reply src/Angor/Shared/DerivationOperations.cs:227]: But later down you await the task is it means the browser will still freeze no?

**Review by dangershony — APPROVED** [review]

**Pull request merged**